src/single_cell/AbstractActionPotentialMethod.cpp

```diff
@@ -358,15 +358,35 @@ OdeSolution AbstractActionPotentialMethod::PerformAnalysisOfTwoPaces(
     // Get voltage properties
     const unsigned voltage_index = pModel->GetSystemInformation()->GetStateVariableIndex("membrane_voltage");
     std::vector<double> voltages = solution.GetVariableAtIndex(voltage_index);
-    CellProperties voltage_properties(voltages, solution.rGetTimes(),
-                                      mActionPotentialThreshold);
+    std::vector<double> times = solution.rGetTimes();
+
 
     std::vector<double> apd90s;
     std::vector<double> peak_voltages;
     // See if we can get back some action potential duration(s).
     try
     {
-        apd90s = voltage_properties.GetAllActionPotentialDurations(90);
+
+        // split into num_paces_to_analyze paces for analysis
+        unsigned current_index = 0;
+        std::vector<CellProperties> voltage_properties;
+        for (unsigned pace = 0; pace < num_paces_to_analyze; pace++)
+        {
+            std::vector<double> pace_voltages;
+            std::vector<double> pace_times;
+            const double pace_end_time = (pace + 1) * s1_period;
+            while (current_index < times.size() && times[current_index] <= pace_end_time)
```

Comment:
can we do something a bit tidier with a range constructor for the `std::vector<double> pace_<things>(some element of full vector, some other element of full vector)` if something like that exists by C++17?

Reply:
? Shouldn't need any search, we know exactly what these indices are ahead of time, it's a regularly sampled vector isn't it, given by the `printingTimeStep`, not just the CVODE steps I think? But might have forgotten....